### ISSUE TITLE
API: Consider Using Graphql

### ISSUE THREAD

**Issue by crstamps2** [homedepot/infinite-wish-board#45]
Proposal to add [FEATURE] to to the site. 

**Feature Name** 
Consider using Graphql

**Functionality**
Graphql is great for getting all data needed in a single payload. Combined with an authorization library, it would be easy to layout authorization role based access, without us having to do a lot of custom controller work.

**Notes**

https://graphql.org/graphql-js/running-an-express-graphql-server/

**Comment by alex-hall** [comment]
I think we're open to this, as long as it's actually solving a concrete problem for the project.

**Comment by alex-hall** [comment]
How much work would it take to convert existing endpoints?

**Comment by fernandocortez** [comment]
Don't know how much of the front end has already been created, but we could leverage GraphQL to feed data into a Gatsby project. The visual components are also React.

**Comment by crstamps2** [comment]
I am going to dump some links here as I come across them. Things that I think we could use that would help us with some of our goals. I will edit this comment as I go. 

https://levelup.gitconnected.com/realtime-subscriptions-with-graphql-and-nodejs-fc27898e1c14
This allows a system to subscribe to database changes. I think this would be cool for our wish wall.

**Comment by alex-hall** [comment]
I've seen some work on this @crstamps2 , how is it coming? Need anything?

**Comment by aclaire-thd** [comment]
Of course, we'd have to update the UI to take advantage of the new GraphQL capabilities...
